### PR TITLE
Internal: project.yaml related fixes

### PR DESCRIPTION
**Pull request by Subbarker**
## Description

1. Install python yaml library on test image
2. remove `local` from bash var to stop hiding errors within python exec
3. refactor project.yaml to use yaml lists

## Checklist:
- Unit tests
  - [x] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [x] Integration tests do not apply.
  - [ ] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: kokoro/scripts/test/go_test.sh [hunk 1]
```diff
@@ -86,7 +86,7 @@ function set_platforms() {
     return 1
   fi
   # At minimum, PLATFORMS will be the distros from "representative" for TARGET/ARCH in projects.yaml.
-  local platforms=$(yaml project.yaml "['targets']['${TARGET}']['architectures']['${ARCH}']['test_distros']['representative']")
+  platforms=$(yaml project.yaml "['targets']['${TARGET}']['architectures']['${ARCH}']['test_distros']['representative']")
```
Comment (martijnvans) [inline kokoro/scripts/test/go_test.sh:89]: you can do this:

```
local platforms
platforms=$(stuff)
```

File: project.yaml [hunk 2]
```diff
@@ -4,146 +4,146 @@ targets:
...
         test_distros:
           representative:
-            ubuntu-minimal-2004-lts-arm64
+          - ubuntu-minimal-2004-lts-arm64
           exhaustive:
-            ubuntu-2004-lts-arm64
+          - ubuntu-2004-lts-arm64
   jammy:
     architectures:
       x86_64:
```
Comment (martijnvans) [inline project.yaml:53]: should architectures be in a list too? seems like yes to me, but I dunno

File: kokoro/scripts/test/go_test.sh (hunk 1 above)
Comment (Subbarker) [reply kokoro/scripts/test/go_test.sh:89]: Done

File: project.yaml (hunk 2 above)
Comment (Subbarker) [reply project.yaml:53]: Lists can't be indexed by key, only integers. this would break the logic we have.

**Review by martijnvans — APPROVED** [review]

**Pull request merged**